client/lib/create-selector/index.js

```diff
@@ -105,3 +106,28 @@ export default function createSelector( selector, getDependants = DEFAULT_GET_DE
 	}, { memoizedSelector } );
 }
 
+export function createSelectorPerKey( selector, getDependants = DEFAULT_GET_DEPENDANTS, getCacheKey = DEFAULT_GET_CACHE_KEY ) {
```

Comment:
You right, I've pushed commit with tests for the new method. I'm using the existing test suite of `createSelector` and added test specific for the new method.